Makefile

```diff
@@ -167,3 +169,84 @@ endif
 # Help documentatin à la https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
 help:
 	@cat Makefile | grep -v '\.PHONY' |  grep -v '\help:' | grep -B1 -E '^[a-zA-Z_.-]+:.*' | sed -e "s/:.*//" | sed -e "s/^## //" |  grep -v '\-\-' | sed '1!G;h;$$!d' | awk 'NR%2{printf "\033[36m%-30s\033[0m",$$0;next;}1' | sort
+
+# server-debug builds and deploys a debug version of the plugin for your architecture.
+# Then resets the plugin to pick up the changes.
+.PHONY: debug
+debug: server-debug reset
+
+.PHONY: server-debug
+server-debug:
+
+	./build/bin/manifest apply
+	mkdir -p server/dist
+
+ifeq ($(OS),Darwin)
+	cd server && env GOOS=darwin GOARCH=amd64 $(GO) build -gcflags "all=-N -l" -o dist/plugin-darwin-amd64;
+else ifeq ($(OS),Linux)
+	cd server && env GOOS=linux GOARCH=amd64 $(GO) build -gcflags "all=-N -l" -o dist/plugin-linux-amd64;
+else ifeq ($(OS),Windows_NT)
+	cd server && env GOOS=windows GOARCH=amd64 $(GO) build -gcflags "all=-N -l" -o dist/plugin-windows-amd64.exe;
+else
+	$(error make debug depends on uname to return your OS. If it does not return 'Darwin' (meaning OSX), 'Linux', or 'Windows_NT' (all recent versions of Windows), you will need to edit the Makefile for your own OS.)
+endif
+
+	rm -rf dist/
+	mkdir -p dist/$(PLUGIN_ID)/server/dist
+	cp $(MANIFEST_FILE) dist/$(PLUGIN_ID)/
+	cp -r server/dist/* dist/$(PLUGIN_ID)/server/dist/
+	mkdir -p ../mattermost-server/plugins
+	cp -r dist/* ../mattermost-server/plugins/
+
+# webapp-debug builds and deploys a debug version of the plugin's webapp
+.PHONY: webapp-debug
+webapp-debug:
+
+ifneq ($(HAS_WEBAPP),)
+# link the webapp directory
+	mkdir -p ../mattermost-server/plugins/$(PLUGIN_ID)/webapp
```

Comment:
I did a quick search and it seems anything like that is related to hooking into node, not really running shell commands. (at least what I could find)
For now I'm fine with pressing up and enter to do a `make reset` -- it's faster than what we're doing now. :)

Reply:
Right, but you can invoke any shell commands you need from `node`. Thinking something like https://www.npmjs.com/package/on-build-webpack that kicks off a `make deploy` or whatever.

Pushing back on this mostly because this isn't just code for us to use temporarily -- this is /the/ example for how to develop Mattermost plugins. People will fork this repository and likely be stuck with whatever tooling we give them for some time (not everyone updates against mattermost-plugin-sample regularly).

It would be terrific if `make webapp-debug` launched the webpack process, listened for compilation events, and effectively triggered a `BUILD_FAST=1 make deploy` that, in turn, reloaded the plugin client-side. You wouldn't even need to reload your browser in that case :)